### ISSUE TITLE
KeyboardProvider , UI bug after device rotate , using keyboard-controller and react-navigation

### ISSUE THREAD

**Issue by term51** [kirillzyusko/react-native-keyboard-controller#547]
**Describe the bug**
When I finished the UI I decided to test the display with a horizontal screen and noticed that the UI shifts depending on the screen rotation. Initially, I didn't use the react-native-safe-area-context component, when I figured out what the problem was I saw that your example used it, I put it in but it didn't work. I don't know what to do with it, I tried removing components from the list one by one, but the problem disappeared only after removing KeyboardProvider. At this point, the only option is to remove your library and consider how to customize the Keyboard from react-native to make it work properly. If you need additional information, please reply as soon as possible.

versions of some of the usable libraries
    "react-native-reanimated": "3.15.0",
    "react-native-paper": "5.12.5",
    "@react-navigation/bottom-tabs": "6.6.1",
    "@react-navigation/native": "6.1.18",
    "@react-navigation/native-stack": "6.11.0",
    "@react-navigation/stack": "6.4.1",
    
    
**Code snippet**
```tsx
return (
    <SafeAreaProvider initialMetrics={initialWindowMetrics}>
      <KeyboardProvider >
        <PaperProvider theme={selectedTheme}>
          <NotificationToastProvider>
            <ConfirmationDialogProvider>
              <Portal.Host>
                <Navigation />
              </Portal.Host>
            </ConfirmationDialogProvider>
          </NotificationToastProvider>
        </PaperProvider>
      </KeyboardProvider>
    </SafeAreaProvider>
  );
```
**To Reproduce**

1. Set up according to the instruction
2. Click rotate on the emulator
3. See the problem

**Expected behavior**
bottom and top are formatted to fit the width of the screen

**Screenshots**
Normal
![image](https://github.com/user-attachments/assets/d1cd8bbb-5916-4fec-a46e-8220241a0170)

Rotated
![image](https://github.com/user-attachments/assets/f7b3b227-5291-4972-915f-0360c7d090f2)

**Smartphone (please complete the following information):**

- Desktop OS:  Windows 11
- Device: Pixel 8, Samsung m21
- OS: Android 7, 12, 14
- RN version:  0.74.0
- RN architecture: old
- JS engine:  Hermes
- Library version: 1.12.7

**Comment by kirillzyusko** [comment]
Also tested bottom tabs in example app (on API 28) and it seems to be working fine - let me know if I'm doing something wrong:

<img width="631" alt="Screenshot 2024-08-14 at 00 09 01" src="https://github.com/user-attachments/assets/f7830fea-0f34-45ec-8f31-017b295184b6">

**Comment by term51** [comment]
Okay, I'm doing it now.

**Comment by term51** [comment]
https://github.com/term51/for-fix-ui-bug

The problem remains, by the way, the shift is relative to the turn.

![image](https://github.com/user-attachments/assets/3c750545-c678-4d49-a2c7-940343949740)

**Comment by term51** [comment]
No problem, it can be wait, thank you for such a quick response

**Comment by term51** [comment]
Я перевый раз пробую все это завести на ios, и могу где-то ошибаться. вродебы приложение и билдится, но появляется такая ошибка и ui смещается по другому.


/Users/user/Library/Developer/Xcode/DerivedData/SanctumMobileApp-einpcbqzexherngdvjhbypuodtew/Build/Intermediates.noindex/SanctumMobileApp.build/Debug-iphonesimulator/SanctumMobileApp.build/DerivedSources/SanctumMobileApp_vers.c 

module map file '/Users/user/Library/Developer/Xcode/DerivedData/SanctumMobileApp-einpcbqzexherngdvjhbypuodtew/Build/Products/Debug-iphonesimulator/react-native-keyboard-controller/react_native_keyboard_controller.modulemap' not found

Содержимое файла
 extern const unsigned char SanctumMobileAppVersionString[];
 extern const double SanctumMobileAppVersionNumber;

 const unsigned char SanctumMobileAppVersionString[] __attribute__ ((used)) = "@(#)PROGRAM:SanctumMobileApp  PROJECT:SanctumMobileApp-1" "\n";
 const double SanctumMobileAppVersionNumber __attribute__ ((used)) = (double)1.;

**Comment by kirillzyusko** [comment]
@term51 can you please check if https://github.com/kirillzyusko/react-native-keyboard-controller/pull/550 fixes the problem?

**Comment by kirillzyusko** [comment]
The fix is published under `1.13.1`! Let me know if the issue is still there so that I can re-open this issue!
